jquery.mentions.coffee

```diff
@@ -237,8 +243,14 @@ class MentionsInput extends MentionsBase
         @_updateValue()
 
     _createHidden: ->
-        hidden = $('<input>', {type: 'hidden', name: @input.attr('name')})
+        hidden = $('<input>', {type: 'hidden', name: @input.attr('name')});
+
+        $.each(@input.data(), (name, value) ->    hidden.attr("data-" + name.replace(/([a-zA-Z])(?=[A-Z])/g, '$1-').toLowerCase(), JSON.stringify(value))
```

Comment:
@ivirabyan  : I tried this bit before and strangely it does not work.
`hidden.data(@input.data())`
this actually copies the data attributes on the jquery object instead of the dom and for some reason they are not visible to jquery validate.
have a look at this http://stackoverflow.com/questions/17667732/data-attribute-value-updated-by-jquery-is-not-visible-in-dom

The code i added is workaround to make sure that the attributes are visible to jquery validate.